src/bootstrap/tool.rs

```diff
@@ -665,6 +665,7 @@ tool_extended!((self, builder),
         });
         self.extra_features.push("clippy".to_owned());
     };
+    RustAnalyzer, rust_analyzer, "src/tools/rust-analyzer/crates/rust-analyzer", "rust-analyzer", {};
```

Comment:
Why isn't this `src/tools/rust-analyzer` which is similar to the rest?

Reply:
rust-analyzer is a workspace (with a virtual manifest), the actual
binary is provide by `crates/rust-analyzer` package

Reply:
Because the crate that must be compiled is found at `src/tools/rust-analyzer/crates/rust-analyzer`, not `src/tools/rust-analyzer/`.

Reply:
@bjorn3 Yes, I understand that. But should it keep the consistency with the other binaries as well? I mean it may be extra work but does it worth it?

Reply:
I think best structure for each tool is more important than common structure. The current structure of ra works great imo. Moving the lsp server to the top level would make it worse. One of the core design principles of ra is to treat lsp as a not very important thin shim over the internal incremental compiler.